lib/galaxy/quota/__init__.py

```diff
@@ -174,6 +185,97 @@ def get_quota(self, user, quota_source_label=None) -> Optional[int]:
             else:
                 return None
 
+    def relabel_quota_for_dataset(self, dataset, from_label: Optional[str], to_label: Optional[str]):
+        adjust = dataset.get_total_size()
+        with_quota_affected_users = """WITH quota_affected_users AS
+(
+    SELECT DISTINCT user_id
+    FROM history
+        INNER JOIN
+            history_dataset_association on history_dataset_association.history_id = history.id
+        INNER JOIN
+            dataset on history_dataset_association.dataset_id = dataset.id
+    WHERE
+        dataset_id = :dataset_id
+)"""
+        engine = object_session(dataset).bind
```

Comment:
@jmchilton I missed this one: starting SA 1.4, to get the engine, `get_bind()` should be used (like line 180). No worries - I'll fix this in the SA2.0 PR. So just FYI.